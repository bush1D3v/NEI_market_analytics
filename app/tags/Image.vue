<script lang="ts" setup>
import { computed, onMounted, ref, type HTMLAttributes, type ImgHTMLAttributes } from "vue";
import { cn } from "@/lib/utils";
import Skeleton from "@/components/ui/skeleton/Skeleton.vue";
import ImageSkeletonSvg from "@/components/Svg/ImageSkeletonSvg.vue";

export interface ImageProps extends /* @vue-ignore */ ImgHTMLAttributes {
    class?: HTMLAttributes[ "class" ];
    src: string;
    alt: string;
    width: string;
    height: string;
}

const props = defineProps<ImageProps>();

const loading = ref<boolean>(true);
const visible = ref<boolean>(true);

function onImageLoad() {
    loading.value = false;
}

const roundedClass = computed(() => {
    const classList = props.class?.split(" ") || [];
    return classList.filter((cls: string) => cls.startsWith("rounded")).join(" ");
});

onMounted(() => {
    setTimeout(() => {
        if (loading.value) {
            visible.value = false;
        }
    }, 5000);
});
</script>

<template>
    <div v-if="visible" class="relative" :class="`w-[${props.width}px] h-[${props.height}px] flex items-center`">
        <Skeleton v-if="loading" :class="`w-full h-full absolute ${roundedClass}`">
            <ImageSkeletonSvg />
        </Skeleton>
        <img :src="props.src" :alt="props.alt" :width="props.width" :height="props.height" :class="cn(props.class)"
            data-testid="Image" @load="onImageLoad()" />
    </div>
    <div v-else :class="` flex items-center justify-center`">
        <ImageSkeletonSvg :class="`w-[${props.width}px] h-[${props.height}px]`" />
    </div>
</template>
