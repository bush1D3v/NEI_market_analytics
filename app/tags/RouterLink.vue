<script lang="ts" setup>
import type {HTMLAttributes} from "vue";
import {cn} from "@/lib/utils";

export interface RouterLinkProps {
	class?: HTMLAttributes["class"];
	to: string;
}

const props = defineProps<RouterLinkProps>();
</script>

<template>
    <router-link :to="props.to" :class="cn('duration-150 ease-in-out cursor-pointer', props.class)"
        data-testid="RouterLink">
        <slot />
    </router-link>
</template>
