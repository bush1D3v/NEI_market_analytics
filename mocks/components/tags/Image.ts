import type {ImageProps} from "@/tags/Image.vue";

export const ImageMock: ImageProps = {
	alt: "image",
	height: "80",
	width: "80",
	src: "https://via.placeholder.com/80",
	class: "w-20 h-20 rounded-full",
};
