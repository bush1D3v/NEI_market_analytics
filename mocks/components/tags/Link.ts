import type {LinkProps} from "@/tags/Link.vue";

export const LinkMock: LinkProps = {
	href: "https://www.google.com",
	target: "_blank",
	class: "font-bold underline",
};
