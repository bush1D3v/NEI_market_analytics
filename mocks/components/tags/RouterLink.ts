import type {RouterLinkProps} from "@/tags/RouterLink.vue";

export const RouterLinkMock: RouterLinkProps = {
	to: "/about-us",
	class: "font-bold underline",
};
