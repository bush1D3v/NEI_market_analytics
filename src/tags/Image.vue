<script lang="ts" setup>
import { computed, ref, type HTMLAttributes, type ImgHTMLAttributes } from "vue";
import { cn } from "@/lib/utils";
import Skeleton from "@/components/ui/skeleton/Skeleton.vue";
import ImageSkeletonSvg from "@/components/Svg/ImageSkeletonSvg.vue";

export interface ImageProps extends /* @vue-ignore */ ImgHTMLAttributes {
    class?: HTMLAttributes[ "class" ];
    src: string;
    alt: string;
    width: string;
    height: string;
}

const props = defineProps<ImageProps>();

// Data
const loading = ref<boolean>(true);

// Methods
function onImageLoad() {
    loading.value = false;
}

const roundedClass = computed(() => {
    const classList = props.class?.split(" ") || [];
    return classList.filter((cls: string) => cls.startsWith("rounded")).join(" ");
});
</script>

<template>
    <div class="relative" :class="`w-[${props.width}px] h-[${props.height}px]`">
        <Skeleton v-if="loading" :class="`w-full h-full absolute ${roundedClass}`">
            <ImageSkeletonSvg />
        </Skeleton>
        <img :src="props.src" :alt="props.alt" :width="props.width" :height="props.height" :class="cn(props.class)"
            data-testid="Image" @load="onImageLoad()" />
    </div>
</template>
