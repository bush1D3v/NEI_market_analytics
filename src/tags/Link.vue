<script lang="ts" setup>
import type {AnchorHTMLAttributes, HTMLAttributes} from "vue";
import {cn} from "@/lib/utils";

export interface LinkProps extends /* @vue-ignore */ AnchorHTMLAttributes {
	class?: HTMLAttributes["class"];
	target: "_blank" | "_self" | "_parent" | "_top";
	href: string;
}

const props = defineProps<LinkProps>();
</script>

<template>
    <a :href="props.href" :target="props.target" rel="no-referrer" :class="cn(props.class)" data-testid="Link">
        <slot />
    </a>
</template>
