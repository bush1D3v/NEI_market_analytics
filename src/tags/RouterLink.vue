<script lang="ts" setup>
import type { HTMLAttributes } from "vue";
import { cn } from "@/lib/utils";

export interface LinkProps {
    class?: HTMLAttributes[ "class" ];
    to: string;
}

const props = defineProps<LinkProps>();
</script>

<template>
    <router-link :to="props.to" :class="cn('duration-150 ease-in-out cursor-pointer', props.class)"
        data-testid="RouterLink">
        <slot />
    </router-link>
</template>
