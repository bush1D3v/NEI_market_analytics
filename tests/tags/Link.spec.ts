import { test, expect, describe } from "vitest";
import Link from "@/tags/Link.vue";
import { LinkMock } from "../../mocks/components/tags/Link";
import { mount } from "@vue/test-utils";

describe("Link Component Tests", () => {
    const wrapper = mount(Link, {
        props: { ...LinkMock }, slots: {
            default: 'Texto do Link'
        }
    });
    const linkComponent = wrapper.findComponent(Link);

    test("Should be able to render the component correctly", async () => {
        expect(linkComponent).toBeDefined();
    });

    test("Should be able to snapshot the component correctly", async () => {
        expect(linkComponent.html()).toMatchSnapshot();
    });

    test("Should be able to render the component with the correct props", async () => {
        expect(linkComponent.props("href")).toBe(LinkMock.href);
        expect(linkComponent.props("target")).toBe(LinkMock.target);
        expect(linkComponent.props("class")).toBe(LinkMock.class);
    });

    test("Should be able to render the component with the correct rel", async () => {
        expect(linkComponent.attributes("rel")).toBe("no-referrer");
    });

    test("Should be able to render the component with the correct text", async () => {
        expect(linkComponent.text()).toBe("Texto do Link");
    });

    test("Should be able to hover the component", async () => {
        expect(linkComponent.trigger("mouseenter")).toBeDefined();
    });
});
