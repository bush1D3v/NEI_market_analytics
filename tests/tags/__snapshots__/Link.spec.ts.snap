// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Link Component Tests > Should be able to snapshot the component correctly 1`] = `"<a href="https://www.google.com" target="_blank" rel="no-referrer" class="hover:opacity-50 duration-150 ease-in-out font-bold underline" data-testid="Link">Texto do Link</a>"`;
